### DISCUSSION TITLE
Automation failed

### DISCUSSION THREAD

**Discussion by suchoi31** [svenerbe/ecoflow_dynamic_power_adjustment#38] [Q&A]
Hello and good evening,

I was trying exactly the example which was mentioned in the tutorial for the YAML script regarding the HA Automation:

>>
service: pyscript.set_ef_powerstream_custom_load_power
data:
  SerialNumber: Your Serial Nr.
  TotalPower: "{{ states('sensor.your sensor') | round }}"
  Automation: true
<<

Unfortunately, I get an issue, see picture
<img width="541" alt="2024-11-08 23_01_11-Einstellungen – Home Assistant und 4 weitere Seiten - Persönlich – Microsoft​ Ed" src="https://github.com/user-attachments/assets/d5e60f09-142a-4046-b5e5-75149cc6f07f">

--> Error message: "Message malformed: extra keys not allowed @ data['service']"

What is wrong here?

Thank you in advance for any help.

**Comment by Just-Zuul** [comment]
TRY:

action: pyscript.set_ef_powerstream_custom_load_power

if you have a HA 2024.10.x or later
